### PR TITLE
Upgrade Aspen to 0.42

### LINKED ISSUE #766: Upgrade aspen to the last released version (0.42)
This website is still using `aspen 0.37` while the last release is `0.42`. It introduced interesting security improvements/fixes, so it's useful.

[I asked @whit537 to do it](https://github.com/gratipay/inside.gratipay.com/issues/760#issuecomment-238564481) since… yeah, it's his framework and I'm sure he'll have to spend way less time than me on this task. Thanks! 💃

### PR DESCRIPTION
**Pull request by whit537**
Closes #766. Ready for review, @nashe @techtonik! :-)

### CODE REVIEW

**Comment by nashe** [comment]
!m @whit537 

I was trying to upgrade it myself after @techtonik's comment on the issue (he's completely right), I should have told it :-( 

Anyway, some code related to the menu is broken (cf Travis) and I was writing [pages tests like it's done for gratipay.com](https://github.com/gratipay/gratipay.com/blob/master/tests/py/test_pages.py) so we don't have to ensure manually that all the pages are still working after one upgrade. I'll work on what's left 🐱

**Comment by whit537** [comment]
Gah, sorry. I'm happy to take a look when I get a chance if you don't get to it first.

**Comment by whit537** [comment]
Rebased, was 1b535ba7d23e4395ff8437333f61466b00ea21e5.

**Comment by whit537** [comment]
This is ready for review again @mattbk @techtonik @JessaWitzel et al.

File: tests/py/test_security.py [hunk 1]
```diff
@@ -2,10 +2,13 @@
 from pytest import yield_fixture
 
 from aspen.testing.client import Client
+from inside_gratipay.main import website
 
 @yield_fixture
 def client():
-    yield Client(www_root='www', project_root='')
+    client = Client(www_root='www', project_root='.')
+    client._website = website
```
Comment (JessaWitzel) [inline tests/py/test_security.py]: What is this for?

Comment (whit537) [inline tests/py/test_security.py]: We're telling Aspen's testing infrastructure to use the `website` that we are making over in `inside_gratipay/main.py`, which is configured how we want it.

Under previous versions of Aspen, Aspen would look in the project root for a magically named file ("`configure-aspen.py`") and take configuration from there. The magic is gone in 0.42.

**Comment by whit537** [comment]
Rebased, was 9e4ecafa308e9a554ea164184af47eb7a376ed3a.

**Comment by EdOverflow** [comment]
Has this been deployed?

**Comment by mattbk** [comment]
Looks like: https://github.com/gratipay/gratipay.com/commits/master/www/version.txt

**Comment by whit537** [comment]
@mattbk But this is for Inside Gratipay, not Gratipay.com.

Inside Gratipay is deployed automatically on push to master (except when it breaks #947  ;). So ... yes? :-)

[Confirmed](https://dashboard.heroku.com/apps/inside-gratipay-com):

> Deployed a2f565f

That [includes](https://github.com/gratipay/inside.gratipay.com/commits/a2f565feb5509e00196d0f2489fcb63e7d68447d) this.

screen shot 2016-12-27 at 5 00 40 pm

**Pull request merged**